mlos_bench/mlos_bench/config/schemas/environments/composite-env-subschema.json

```diff
@@ -19,36 +19,44 @@
                 "config": {
                     "type": "object",
                     "$comment": "config objects need to specify their entire schema, so we need to include the unevaluatedProperties handling here.",
-                    "properties": {
-                        "include_children": {
-                            "description": "List of child environment config files to include in the composite environment.",
-                            "type": "array",
-                            "items": {
-                                "type": "string",
-                                "pattern": "[.]json[c]?$"
-                            },
-                            "minItems": 1
-                        },
-                        "children": {
-                            "description": "List of inline child environment configs to include in the composite environment.",
-                            "type": "array",
-                            "items": {
-                                "$comment": "Each config included can be a leaf environment config.  We do not currently support inline nested composite environments.",
-                                "$ref": "./leaf-environment-subschemas.json",
-                                "unevaluatedProperties": false
-                            },
-                            "minItems": 1
-                        }
-                    },
-                    "unevaluatedProperties": false,
-                    "anyOf": [
+                    "allOf": [
                         {
-                            "required": ["include_children"]
+                            "$ref": "./common-environment-subschemas.json#/$defs/leaf_environment_config"
```

Comment:
Not sure I follow this.  You want to accept a leaf environment config as if it were a composite env?

That should only be happening in the `children` property or the base `service-schema.json`, right?

Reply:
Every environment, including the `CompositeEnv`, can have `tunable_params`, `const_args`, and `required_args` in its `config` section - it is all defined in the base class. I am just reusing the functionality we already have with the `const_args`. A `CompositeEnv` can also have the `tunable_params` section to explicitly specify the tunables; it will add the tunables from the children to the mix

Reply:
Let's rename `leaf_environment_config` to `common_environment_config` then in the `common-environment-subschemas.json` - it'll make it more clear what's going on I think.